### PR TITLE
feat: implement custom querier #lsc-80

### PR DESCRIPTION
**Pull request by NeverHappened**
Implements https://p2pvalidator.atlassian.net/browse/LSC-80

Custom querier that implements two queries:
- interchain query result
- interchain account address
- query registered queries

What's missing:
- maybe add some bash script analogous to osmosis that will download .wasm contracts from github (like https://github.com/osmosis-labs/osmosis/blob/main/wasmbinding/testdata/download_releases.sh) (but we don't have them yet)

### CODE REVIEW

File: testutil/test_helpers.go [hunk 1]
```diff
@@ -2,18 +2,131 @@ package testutil
...
+	coordinator *ibctesting.Coordinator
+
+	// testing chains used for convenience and readability
+	ChainA *ibctesting.TestChain
+	ChainB *ibctesting.TestChain
+
+	Path *ibctesting.Path
+}
+
+func SetupIBCConnection(t *testing.T) *TestIBCConnectionStruct {
```
Comment (NeverHappened) [inline testutil/test_helpers.go]: Not sure here:
I've extracted this from `keeper_test.go`.
But I didn't remove it from there.
I don't know whether I should remove it or not cause it's pretty hardcoded there

File: wasmbinding/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This package allows for custom queries and custom messages sends from contract.
+
+### What supported 
+- Queries:
+  - InterchainQueryResult - Get the result of a registered interchain query by query_id
+  - InterchainAccountAddress - Get the interchain account address by owner_id and connection_id
+
+- Messages: TODO
```
Comment (NeverHappened) [inline wasmbinding/README.md:9]: Messages are being implemented in other task

File: wasmbinding/test/custom_query_test.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	"io/ioutil"
+	"testing"
+)
+
+func init() {
+	config := app.GetDefaultConfig()
+	config.Seal()
+}
+
+func TestInterchainQueryResult(t *testing.T) {
```
Comment (NeverHappened) [inline wasmbinding/test/custom_query_test.go]: Both tests need to check unhappy paths too, I wasn't on time to do it

File: testutil/test_helpers.go (hunk 1 above)
Comment (NeverHappened) [inline testutil/test_helpers.go]: Not sure that all added stuff belongs here. Maybe extract to another helper to keep things clean?

File: wasmbinding/README.md (hunk 2 above)
Comment (foxpy) [reply wasmbinding/README.md:9]: Is it a good idea for me to rebase my work on your PR after we get your PR merged?

Comment (NeverHappened) [reply wasmbinding/README.md:9]: Yeah, we can combine it pretty easily I think

File: app/app.go [hunk 4]
```diff
@@ -501,7 +502,8 @@ func New(
 	// The last arguments can contain custom message handlers, and custom query handlers,
 	// if we want to allow any custom callbacks
 	supportedFeatures := "iterator,staking,stargate"
```
Comment (mmikev) [inline app/app.go]: We need `neutron` feature here (and on smart-contracts side too). See https://github.com/neutron-org/neutron/pull/22/files#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7R506 and https://github.com/neutron-org/neutron-contracts/blob/feat/msgs-bindings/packages/bindings/src/lib.rs#L10 for reference

**Review by pr0n00gler — DISMISSED** [review]
The rest of the PR looks good

File: wasmbinding/queries.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package wasmbinding
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/neutron-org/neutron/x/interchainqueries/types"
+	icatypes "github.com/neutron-org/neutron/x/interchaintxs/types"
+)
+
+// GetInterchainQueryResult is a function, not method, so the message_plugin can use it
```
Comment (zavgorodnii) [inline wasmbinding/queries.go]: not sure what this comment means tbh

**Review by zavgorodnii — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**